### PR TITLE
[Implementacion JUnit] - Implementation JUnit

### PR DESCRIPTION
**Pull request by AleFalcon**
## Summary
Implementatio JUnit, with creation test for user and books controllers, and model.

## Trello Card
https://trello.com/c/jbkhpnh6/28-junit-testing

### CODE REVIEW

**Review by Gianni-Nicolas — DISMISSED** [review]

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -216,4 +219,44 @@ public void update (String genre, String author, String image, String title, Str
         this.setPage(page);
     }
 
+    public void setAtributes(String genre, String author, String image, String title, String subtitle, String publisher, String year, Integer page, String isbn){
+        this.setIsbn(isbn);
+        this.update(genre, author, image, title, subtitle, publisher, year, page);
+    }
+
+    public boolean equal_book (Book book){
```
Comment (facucompe) [inline src/main/java/wolox/training/models/Book.java]: You should overrride the method `equals()`

File: src/test/java/wolox/training/controllers/BookTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        ));
+  }
+
+  @Test
+  public void whenCreateBook_thenBookIsReturned() throws Exception{
+    Mockito.when(mockedBookRepository.save(oneTestBook))
+        .thenReturn((oneTestBook));
+    mvc.perform(post(VariablesConstants.BOOK_URL)
+        .contentType(MediaType.APPLICATION_JSON)
+        .content("{\"genre\": \"Terror\","
```
Comment (facucompe) [inline src/test/java/wolox/training/controllers/BookTest.java]: Try to use an object mapper instead https://mkyong.com/java/jackson-2-convert-java-object-to-from-json/ .
Or use jsonPath https://mkyong.com/spring-boot/spring-test-how-to-test-a-json-array-in-jsonpath/

File: src/test/java/wolox/training/controllers/BookTest.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+import wolox.training.utils.VariablesConstants;
+
+@RunWith(MockitoJUnitRunner.class)
+@WebMvcTest(BookController.class)
+public class BookTest {
```
Comment (facucompe) [inline src/test/java/wolox/training/controllers/BookTest.java]: `BookControllerTest`

File: src/test/java/wolox/training/controllers/UserTest.java [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+import org.springframework.util.LinkedMultiValueMap;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+import wolox.training.utils.VariablesConstants;
+
+@RunWith(MockitoJUnitRunner.class)
+@WebMvcTest(UserController.class)
+public class UserTest {
```
Comment (facucompe) [inline src/test/java/wolox/training/controllers/UserTest.java]: `UserControllerTest`

File: src/test/java/wolox/training/models/BookTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  private TestEntityManager entityManager;
+
+  @Autowired
+  private BookRepository bookRepository;
+
+  @Test
+  public void whenFindByIsbn_thenReturnBook() {
+    // given
+    Book book = new Book("Terror", "Yo Soy El Autor", "Mi Imagen", "Mi Super Titulo", "Mi SubTitulo", "Publicador", "2020", 3,"99");
+    bookRepository.save(book);
```
Comment (facucompe) [inline src/test/java/wolox/training/models/BookTest.java]: Use `entityManager` instead. 
```
entityManager.persist(book);
entityManager.flush();
```

Comment (facucompe) [reply src/test/java/wolox/training/models/BookTest.java]: You can move it to a method with `@Before`

**Review by facucompe — CHANGES_REQUESTED** [review]

**Review by AleFalcon — COMMENTED** [review]
Corrections

File: src/test/java/wolox/training/models/UserModelTest.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  @Autowired
+  private UserRepository userRepository;
+
+  @Test
+  public void whenFindByUsername_thenReturnUser() {
+    // given
+    User user = new User("Alex", "Alito21", LocalDate
+        .of(1994, 10, 25));
+    userRepository.save(user);
```
Comment (facucompe) [inline src/test/java/wolox/training/models/UserModelTest.java]: You should use entityManager here too

**Review by facucompe — CHANGES_REQUESTED** [review]

**Review by AleFalcon — COMMENTED** [review]
Ready

**Review by facucompe — APPROVED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Pull request merged**